packages/migrate/src/__tests__/DbPull/postgresql-views.test.ts

```diff
@@ -0,0 +1,665 @@
+      tree?.sort((a, b) => a.localeCompare(b, 'en-US'))
+      expect(tree).toMatchInlineSnapshot(`
+        [
+          views/extraneous-file.sql,
+          views/public/simpleuser.sql,
+          views/README,
+          views/work/workers.sql,
+        ]
+      `)
+    })
+  })
+
+  describe('no preview', () => {
+    const { fixturePath } = setupPostgresForViewsIO('no-preview')
+
+    test('basic introspection', async () => {
+      ctx.fixture(path.join(fixturePath))
+
+      const introspect = new DbPull()
+      const result = introspect.parse([])
+      await expect(result).resolves.toMatchInlineSnapshot(``)
+
+      const list = await ctx.fs.listAsync('views')
+      expect(list).toMatchInlineSnapshot(`undefined`)
+
+      const tree = await ctx.fs.findAsync({ directories: false, files: true, recursive: true, matching: 'views/**/*' })
+      expect(tree).toMatchInlineSnapshot(`[]`)
+
+      expect(ctx.mocked['console.log'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(`
+        Prisma schema loaded from schema.prisma
+        Datasource "db": PostgreSQL database "tests-migrate-db-pull-postgresql-views", schemas "public, work" at "localhost:5432"
+      `)
+      expect(ctx.mocked['console.error'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['process.stdout.write'].mock.calls.join('\n')).toMatchInlineSnapshot(`
+
+
+        - Introspecting based on datasource defined in schema.prisma
+
+        ✔ Introspected 2 models and wrote them into schema.prisma in XXXms
+              
+        Run prisma generate to generate Prisma Client.
+
+      `)
+      expect(ctx.mocked['process.stderr.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+    })
+
+    test('introspect with already existing files in "views"', async () => {
+      ctx.fixture(path.join(fixturePath))
+
+      await ctx.fs.dirAsync('views/extraneous-dir')
+      await ctx.fs.fileAsync('views/extraneous-file.sql')
+      const extraneousList = await ctx.fs.listAsync('views')
+      expect(extraneousList).toMatchInlineSnapshot(`
+        [
+          extraneous-dir,
+          extraneous-file.sql,
+        ]
+      `)
+
+      const introspect = new DbPull()
+      const result = introspect.parse([])
+      await expect(result).resolves.toMatchInlineSnapshot(``)
+
+      const list = await ctx.fs.listAsync('views')
+      expect(list).toMatchInlineSnapshot(`
+        [
+          extraneous-dir,
+          extraneous-file.sql,
+        ]
+      `)
+
+      const tree = await ctx.fs.findAsync({ directories: false, files: true, recursive: true, matching: 'views/**/*' })
+      expect(tree).toMatchInlineSnapshot(`
+        [
+          views/extraneous-file.sql,
+        ]
+      `)
+    })
+  })
+})
+
+// TODO remove this once ported to prisma-engines
```

Comment:
Actually it's already tested in `re_intro_keeps_the_field_map` https://github.com/prisma/prisma-engines/blob/a9691be293caff3232d970c1150a952b03f7a0fd/migration-engine/introspection-engine-tests/tests/views/postgresql.rs

Reply:
So I removed it now